engine/runtime/src/test/scala/org/enso/compiler/test/pass/resolve/GenerateDocumentationTest.scala

```diff
@@ -0,0 +1,233 @@
+        .asInstanceOf[IR.Module.Scope.Definition.Method.Binding]
+        .body
+        .asInstanceOf[IR.Expression.Block]
+
+      body.expressions.length shouldEqual 1
+      getDoc(
+        body.expressions(0)
+      ) shouldEqual "<html><body><div class=\"doc\" style=\"font-size: 13px;\"><div><div class=\"\"><div class=\"example\"><div class=\"summary\">&lt;&lt;&lt;&lt;&lt;&lt;Do thing&lt;&lt;&lt;&lt;&lt;&lt;</div></div></div></div></div></body></html>"
+      getDoc(
+        body.returnValue
+      ) shouldEqual "<html><body><div class=\"doc\" style=\"font-size: 13px;\"><div><div class=\"\"><div class=\"example\"><div class=\"summary\">&lt;&lt;&lt;&lt;&lt;&lt;Do another thing&lt;&lt;&lt;&lt;&lt;&lt;</div></div></div></div></div></body></html>"
+    }
+
+    "be associated with the type ascriptions" in {
+      implicit val moduleContext: ModuleContext = mkModuleContext
+      val ir =
+        """
+          |method x =
+          |    ## Id
+          |    f : Any -> Any
+          |    f x = x
+          |
+          |    ## Return thing
+          |    f 1
+          |""".stripMargin.preprocessModule.resolve
+      val body = ir
+        .bindings(0)
+        .asInstanceOf[IR.Module.Scope.Definition.Method.Binding]
+        .body
+        .asInstanceOf[IR.Expression.Block]
+
+      body.expressions.length shouldEqual 2
+      body.expressions(0) shouldBe an[IR.Application.Operator.Binary]
+      getDoc(
+        body.expressions(0)
+      ) shouldEqual "<html><body><div class=\"doc\" style=\"font-size: 13px;\"><div><div class=\"\"><div class=\"example\"><div class=\"summary\">&lt;&lt;&lt;&lt;&lt;&lt;Id&lt;&lt;&lt;&lt;&lt;&lt;</div></div></div></div></div></body></html>"
+      getDoc(
+        body.returnValue
+      ) shouldEqual "<html><body><div class=\"doc\" style=\"font-size: 13px;\"><div><div class=\"\"><div class=\"example\"><div class=\"summary\">&lt;&lt;&lt;&lt;&lt;&lt;Return thing&lt;&lt;&lt;&lt;&lt;&lt;</div></div></div></div></div></body></html>"
+    }
+  }
+
+  "Documentation in complex type definitions" should {
+    implicit val moduleContext: ModuleContext = mkModuleContext
+    "assign docs to all entities" in {
+      val ir =
+        """
+          |## the type Foo
+          |type Foo
+          |    ## the constructor Bar
+          |    type Bar
+          |
+          |    ## the included Unit
+          |    Unit
+          |
+          |    ## a method
+          |    foo x =
+          |        ## a statement
+          |        IO.println "foo"
+          |        ## the return
+          |        0
+          |""".stripMargin.preprocessModule.resolve
+      val tp = ir.bindings(0).asInstanceOf[IR.Module.Scope.Definition.Type]
+      getDoc(
+        tp
+      ) shouldEqual "<html><body><div class=\"doc\" style=\"font-size: 13px;\"><div><div class=\"\"><div class=\"example\"><div class=\"summary\">&lt;&lt;&lt;&lt;&lt;&lt;the type Foo&lt;&lt;&lt;&lt;&lt;&lt;</div></div></div></div></div></body></html>"
```

Comment:
Please format these properly – use multiline strings and write a simple utility function that then collapses tags into single line. This is completely unreadable otherwise.